crates/bevy_reflect/bevy_reflect_derive/src/serialization.rs

```diff
@@ -0,0 +1,85 @@
+use crate::derive_data::StructField;
+use crate::field_attributes::ReflectIgnoreBehavior;
+use bevy_macro_utils::fq_std::{FQBox, FQDefault};
+use quote::quote;
+use std::collections::HashMap;
+use syn::spanned::Spanned;
+use syn::Path;
+
+type ReflectionIndex = usize;
+
+/// Collected serialization data used to generate a `SerializationData` type.
+pub(crate) struct SerializationDataDef {
+    /// Maps a field's _reflection_ index to its [`SkippedFieldDef`] if marked as `#[reflect(skip_serializing)]`.
+    skipped: HashMap<ReflectionIndex, SkippedFieldDef>,
+}
+
+impl SerializationDataDef {
+    /// Attempts to create a new `SerializationDataDef` from the given collection of fields.
+    ///
+    /// Returns `Ok(Some(data))` if there are any fields needing to be skipped during serialization.
+    /// Otherwise, returns `Ok(None)`.
+    pub fn new(fields: &[StructField<'_>]) -> Result<Option<Self>, syn::Error> {
+        let mut skipped = HashMap::default();
+
+        for field in fields {
+            match field.attrs.ignore {
+                ReflectIgnoreBehavior::IgnoreSerialization => {
+                    skipped.insert(
+                        field.reflection_index.ok_or_else(|| {
+                            syn::Error::new(
+                                field.data.span(),
+                                "internal error: field is missing a reflection index",
+                            )
+                        })?,
+                        SkippedFieldDef::new(field)?,
+                    );
+                }
+                _ => continue,
+            }
+        }
+
+        if skipped.is_empty() {
+            Ok(None)
+        } else {
+            Ok(Some(Self { skipped }))
+        }
+    }
+
+    /// Returns a `TokenStream` containing an initialized `SerializationData` type.
+    pub fn as_serialization_data(&self, bevy_reflect_path: &Path) -> proc_macro2::TokenStream {
+        let fields =
+            self.skipped
+                .iter()
+                .map(|(reflection_index, SkippedFieldDef { default_fn })| {
+                    quote! {(
+                        #reflection_index,
+                        #bevy_reflect_path::serde::SkippedField::new(#default_fn)
+                    )}
+                });
+        quote! {
+            #bevy_reflect_path::serde::SerializationData::new(
+                ::core::iter::IntoIterator::into_iter([#(#fields),*])
+            )
+        }
+    }
+}
+
+/// Collected field data used to generate a `SkippedField` type.
+pub(crate) struct SkippedFieldDef {
+    /// The default function for this field.
+    ///
+    /// This is of type `fn() -> Box<dyn Reflect>`.
+    default_fn: proc_macro2::TokenStream,
+}
+
+impl SkippedFieldDef {
+    pub fn new(field: &StructField<'_>) -> Result<Self, syn::Error> {
```

Comment:
considering this is the only constructor, does this respect `#[reflect(default = "my_func")]`?

it might be helpful to add something to the test [here](https://github.com/bevyengine/bevy/blob/d8edc81ac9b1c2011e6a9b64ca7c58d7f9f26412/crates/bevy_reflect/src/lib.rs#L803C1-L803C1).

Reply:
I'm testing this now and it looks like I forgot to account for this (or at least broke it during a rebase). Nice catch, I'll update!